jdi-light/src/main/java/com/epam/jdi/light/elements/base/JDIBase2.java

```diff
@@ -0,0 +1,525 @@
+        searchRules.clear();
+        searchRules.add(name, rule);
+        return this;
+    }
+    public JDIBase setWebElement(WebElement el) {
+        webElement.setForce(el);
+        return this;
+    }
+    public JDIBase setWebElements(List<WebElement> els) {
+        webElements.setForce(els);
+        return this;
+    }
+
+    public JDIBase setLocator(String locator) {
+        return setLocator(defineLocator(locator));
+    }
+    public JDIBase setLocator(By locator) {
+        if (locator != null) {
+            if (name.isEmpty())
+                name = shortBy(locator, this);
+            this.locator.add(locator, this);
+        }
+        return this;
+    }
+    public JDIBase setFrames(List<By> frames) {
+        if (name.isEmpty() && frames.size() > 0)
+            name = shortBy(frames.get(0));
+        this.locator.add(frames, this);
+        return this;
+    }
+    public By getLocator(Object... args) {
+        initContext();
+        return locator.getLocator(args);
+    }
+    public List<By> getFrames() { return locator.getFrames(); }
+
+    protected int timeout = -1;
+    @Override
+    public IBaseElement waitSec(int sec) {
+        timeout = sec;
+        return this;
+    }
+    public int getTimeout() {
+        return timeout > -1 ? timeout : TIMEOUTS.element.get();
+    }
+    public Timer timer() { return new Timer(getTimeout()*1000); }
+
+    @Override
+    public JDIBase setName(String name) {
+        this.name = name;
+        this.varName = name;
+        this.failElement = name;
+        return this;
+    }
+
+    @JDIAction(value = "{0}", timeout = 0)
+    public void visualCheck(String message) { }
+
+    public static final String FAILED_TO_FIND_ELEMENT_MESSAGE
+            = "Can't find Element '%s' during %s seconds";
+    public static final String FIND_TO_MUCH_ELEMENTS_MESSAGE
+            = "Found %s elements instead of one for Element '%s' during %s seconds";
+    public static final String ELEMENTS_FILTERED_MESSAGE
+            = "Found %s elements but none pass results filtering. Please change locator or filtering rules (WebSettings.SEARCH_RULE = )" +
+            LINE_BREAK + "Element '%s' search during %s seconds";
+
+    public WebElement getWebElement() {
+        return get(new Object[]{});
+    }
+    public WebElement get() {
+        WebElement element = get(new Object[]{});
+        VALIDATE_FOUND_ELEMENT.execute(this, element);
+        return element;
+
+    }
+    public static JAction2<JDIBase, WebElement> VALIDATE_FOUND_ELEMENT = JDIBase::validateFoundElement;
+    private static void validateFoundElement(JDIBase base, WebElement element) {
+        for (JFunc1<WebElement, Boolean> rule : base.searchRules().values())
+            if (!rule.execute(element))
+                throw exception("Search rules failed for element. Please check searchRules() for element or in global settings(WebSettings.SEARCH_RULE)");
+    }
+    protected JFunc<WebElement> getElementFunc = null;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5327794)